src/main/java/it/pagopa/pm/gateway/controller/CreditCardPaymentController.java

```diff
@@ -105,10 +101,12 @@ public ResponseEntity<String> resumeCreditCardPaymentStep2(@RequestHeader(requir
                                                                @PathVariable String requestId) {
         log.info("START - POST {}{} info for requestId: {}", REQUEST_PAYMENTS_VPOS, REQUEST_PAYMENTS_RESUME_CHALLENGE, requestId);
         MdcUtils.setMdcFields(mdcFields);
-        String urlRedirect = StringUtils.join(responseUrlRedirect, requestId);
+
         resumeStep2Service.startResumeStep2(requestId);
         log.info("END - POST {}{} info for requestId: {}", REQUEST_PAYMENTS_VPOS, REQUEST_PAYMENTS_RESUME_CHALLENGE, requestId);
-        return ResponseEntity.status(HttpStatus.FOUND).location(URI.create(urlRedirect)).build();
+
+        String vposPollingRedirect = StringUtils.join(vposPollingUrl, requestId);
```

Comment:
Given that both `vposPollingUrl` and `requestId` are expected to be non-null strings you can directly join with `+`
```suggestion
        String vposPollingRedirect = vposPollingUrl + requestId;
```